lib/Backends/NNPI/NNPIOptions.h

```diff
@@ -18,166 +18,306 @@
+  class {                                                                      \
+  public:                                                                      \
+    inline static std::string getName() { return OPT_NAME; }                   \
+    inline static std::string getDesc() { return OPT_DESC; }                   \
+    inline static std::string getEnv() { return OPT_ENV; }                     \
+    inline static std::string getDefault() { return OPT_DEFAULT; }             \
+    inline void setValFromString(std::string v) {                              \
+      val_ = NNPIOptions::getStringAsType<VAR_TYPE>(v);                        \
+    }                                                                          \
+    inline void setVal(VAR_TYPE v) { val_ = v; }                               \
+    inline VAR_TYPE get() const { return (val_); }                             \
+                                                                               \
+    operator VAR_TYPE() const { return val_; }                                 \
+                                                                               \
+  private:                                                                     \
+    VAR_TYPE val_;                                                             \
+  } VAR_NAME;
+
+#define INIT_NNPI_OPTIONS(VAR_NAME, map)                                       \
+  {                                                                            \
+    supportedOptions_[VAR_NAME.getName()] =                                    \
+        llvm::formatv("{0} Default: {1} Environment Variable:",                \
+                      VAR_NAME.getDefault(), VAR_NAME.getEnv())                \
+            .str();                                                            \
+    std::string stVal =                                                        \
+        getFromMap(map, VAR_NAME.getName(), VAR_NAME.getDefault());            \
+    stVal = NNPIOptions::getFromEnv(VAR_NAME.getEnv(), stVal);                 \
+    this->VAR_NAME.setValFromString(stVal);                                    \
+    this->loadedOptions_[VAR_NAME.getEnv()] =                                  \
+        llvm::formatv("{0}", VAR_NAME).str();                                  \
+  }
+
 class NNPIBackendOptions : public NNPIOptions {
 public:
-  explicit NNPIBackendOptions();
-  /// Dump the status of all variables.
-  virtual std::string dumpStatus() const override;
-  /// Lists the supported options (name=>description).
-  static llvm::StringMap<std::string> getSupportedOptions();
   /// Compile to ICE-T if true, ICE-Ref otherwise.
-  bool useIceT;
-  /// Enable execution on device (useIceT_ and !inferOnDevice_ will compile but
+  DECLARE_NNPI_OPTION(useIceT, bool, "UseIceT", "Use inference api",
+                      "USE_ICE_T", "0");
+  /// Enable execution on device (useIceT and !inferOnDevice will compile but
   /// not execute inference).
```

Comment:
`inferOnDevice` implies `useIceT`, correct? Can we expand on some descriptions of what happens here?